===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ PortObjectAddRange:n$3 == { -1 } }) free(*(PortObjectAddRange:poi)) at 3 (line 832, column 4) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
+++ b/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
@@ -829,7 +829,10 @@ int PortObjectAddRange( PortObject * po, int lport, int hport, int not_flag )
    poi->lport = (unsigned short)lport;
    poi->hport = (unsigned short)hport;
 
-   return  sflist_add_tail( po->item_list, poi );
+   int tmp = sflist_add_tail( po->item_list, poi );
+   if (tmp == -1)
+      free(poi);
+   return tmp;
 }
 /*
    Add ANY port
